lib/irc/ConnectionInstance.js

```diff
@@ -355,9 +355,16 @@ ConnectionInstance.create = Promise.coroutine(function*(server, opts, onCreatedC
             log.error(
                 `ConnectionInstance.connect failed after ${connAttempts} attempts (${err.message})`
             );
+
+            if (err.message === "net_error") {
+                // This is the kernel complaining, we should back off.
```

Comment:
Why is the kernel complaining? Why do we back off?

Reply:
This is more of a thing I found when trying to run enormous amounts of IRC connections from my box:
```
{
  "errno": "EBUSY",
  "code": "EBUSY",
  "syscall": "getaddrinfo",
  "hostname": "localhost",
  "host": "localhost",
  "port": 6767
}
```

Reply:
Right, so the host is presumably running out of connections to give? Backing off only makes sense if expect the resource to be available in the future (at this point, it's `EBUSY`). I'm not sure we can expect the server to have space for us in 20s, especially if it's your devbox and the only thing consuming the resource *is* the IRC bridge.

Reply:
I'm not sure what else to do, we can't really exit this func without a connected client but equally we can't harass the kernel for a slot if it's saying no.